File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/AddPostPayload.java
package jp.co.pluswing.mysqlsample.sample.payload;

import jp.co.pluswing.mysqlsample.sample.model.Post;

public class AddPostPayload {
    private AddPostPayload(Boolean success, Post post) {
        this.success = success;
        this.post = post;
    }

    public final Boolean success;
    public final Post post;

    public static AddPostPayload fail() {
        return new AddPostPayload(false, null);
    }
    public static AddPostPayload success(Post post) {
        return new AddPostPayload(true, post);
    }
}

File: sample/src/main/java/jp/co/pluswing/mysqlsample/sample/payload/AddPostPayload.kt
package jp.co.pluswing.mysqlsample.sample.payload

import jp.co.pluswing.mysqlsample.sample.model.Post

data class AddPostPayload private constructor(val success: Boolean, val post: Post?) {
    companion object {
        fun fail(): AddPostPayload {
            return AddPostPayload(false, null)
        }

        fun success(post: Post): AddPostPayload {
            return AddPostPayload(true, post)
        }
    }
}